modules/monitors/README.md

```diff
@@ -0,0 +1,79 @@
+## monitors
+
+This module creates Datadog [monitors](https://docs.datadoghq.com/api/latest/monitors/).
+It accepts all the configuration parameters supported by the Datadog Terraform
+resource.
+
+## Usage
+
+The `datadog_monitors` input takes a map of test definitions. You must supply
+the keys for the map, but the values can follow either of two schemas described 
+below.
+
+There are some optional additions to the monitor definition that are not
+part of the Datadog API schema. These are:
+- `force_delete` (boolean): If true, when deleting the monitor, the monitor 
+  will be deleted even if it is referenced by other resources.
+- `validate` (boolean): If false, the monitor will not be validated during
+  the Terraform plan phase.
+- `enabled` (boolean): If false, the monitor will not be created. This is to 
+  allow you to suppress a monitor created through merging configuration snippets.
+
+### Tags
+
+This module provides special handling for tags. The `tags` attribute of a monitor
+definition in the Datadog API and the Terraform resource is a list of strings. 
+However, the Cloud Posse (as well as AWS) default is to use a map of strings
+for tags. This module allows you to provide either a list or a map for the tags.
+
+If you provide a list, it will be used as is. If you provide a map, it will be
+converted to a list of strings in the format `key:value` (or `key` if `value` is
+`null`). If you provide no tag settings at all, not even an empty list or map,
+the module will use the default tags from the [null-label](https://github.com/cloudposse/terraform-null-label/) module.
+
+### API Schema (preferred)
+
+Datadog provides a REST API for managing monitors. We refer to the responses
+to the API requests for monitor definitions (`GET https://api.datadoghq.com/api/v1/monitor/{monitor_id}`) 
+as the "API Schema" for the tests. This should correspond to the documented 
+API "Model" for Monitors plus additional information such as creation date,
+but if the documentation and the API response differ, we use the API response
+as the source of truth.
+
+You can retrieve Monitor definitions from Datadog via the Datadog Web Console.
+Navigate to the monitor you want to retrieve, click the gear icon in the upper
+right (representing "settings"), and select "Export". This will display a JSON
+representation of the monitor definition. You can then click the "Copy" button
+to copy the JSON to the clipboard.
+
+You can then use the `jsondecode()` function in Terraform to convert the JSON
```

Comment:
```suggestion
to copy the JSON to the clipboard.

<details><summary>Example of JSON for monitors</summary>

Here's an example of the JSON generated from the Datadog web console for a monitor.

```json
{
  "name": "anomaly-recovery-test",
  "type": "query alert",
  "query": "avg(last_1d):anomalies(avg:system.load.1{*}, 'agile', 2, direction='both', interval=300, alert_window='last_37m', count_default_zero='true', seasonality='weekly') >= 1",
  "message": "recovery window test",
  "tags": [],
  "options": {
    "thresholds": {
      "critical": 1,
      "critical_recovery": 0,
      "warning": 0.8
    },
    "notify_audit": false,
    "require_full_window": false,
    "notify_no_data": false,
    "renotify_interval": 0,
    "threshold_windows": {
      "trigger_window": "last_37m",
      "recovery_window": "last_1h"
    },
    "include_tags": false,
    "timeout_h": 1,
    "evaluation_delay": 88,
    "new_host_delay": 300,
    "silenced": {}
  },
  "priority": 5,
  "restricted_roles": null
}
```
</details>

You can then use the `jsondecode()` function in Terraform to convert the JSON
```

Reply:
Since we change the type to `any`, I think this is particularly important to convey what the object may look like. Also, so that when copying and pasting it from the Web Console, they can compare it to this expected output.

Reply:
I agree with including an example, but 2 things:

1. We didn't change the type, it was always `any`
2. I'm a little worried that the JSON output does not contain all the fields. It only contains non-default, non-empty fields. I guess I can just note that in the text.